lib/util.js

```diff
@@ -11,19 +11,20 @@ var Debug;
 
 const formatRegExp = /%[sdj%]/g;
 exports.format = function(f) {
+  const args = arguments;
+  const len  = args.length;
+
   if (typeof f !== 'string') {
-    var objects = [];
-    for (var index = 0; index < arguments.length; index++) {
-      objects.push(inspect(arguments[index]));
+    let objects = new Array(len);
+    for (let i = len; i--;) {
```

Comment:
+1.. Good to know!
On Feb 20, 2016 7:41 AM, "Ben Noordhuis" <notifications@github.com> wrote:

> In lib/util.js
> <https://github.com/nodejs/node/pull/4678#discussion_r53552240>:
>
> >    if (typeof f !== 'string') {
> > -    var objects = [];
> > -    for (var index = 0; index < arguments.length; index++) {
> > -      objects.push(inspect(arguments[index]));
> > +    let objects = new Array(len);
> > +    for (let i = len; i--;) {
>
> Incorrecter with every upgrade. My rule of thumb is to prefer let/const
> and fall back to var only when the optimizer complains. It's easy enough to
> check.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/4678/files#r53552240>.
>